spot-ml/src/main/scala/org/apache/spot/lda/SpotLDAWrapper.scala

```diff
@@ -78,61 +73,96 @@ object SpotLDAWrapper {
       .toDF(DocumentName, DocumentNumber)
       .cache
 
-    //Structure corpus so that the index is the docID, values are the vectors of word occurrences in that doc
+    // Structure corpus so that the index is the docID, values are the vectors of word occurrences in that doc
     val ldaCorpus: RDD[(Long, Vector)] =
       formatSparkLDAInput(docWordCountCache,
         documentDictionary,
         wordDictionary,
         sqlContext)
 
+    val corpusSize = ldaCorpus.count()
+
     docWordCountCache.unpersist()
 
-    //Instantiate optimizer based on input
-    val optimizer = new EMLDAOptimizer
+    // Instantiate optimizer based on input
+    val ldaOptimizer = ldaOptimizerOption match {
+      case "em" => new EMLDAOptimizer
+      case "online" => new OnlineLDAOptimizer().setOptimizeDocConcentration(true).setMiniBatchFraction({
+        if (corpusSize < 2) 0.75
+        else (0.05 + 1) / corpusSize
+      })
+      case _ => throw new IllegalArgumentException(
+        s"Invalid LDA optimizer $ldaOptimizerOption")
+    }
 
     logger.info(s"Running Spark LDA with params alpha = $ldaAlpha beta = $ldaBeta " +
-      s"Max iterations = $maxIterations Optimizer = em")
-    //Set LDA params from input args
+      s"Max iterations = $maxIterations Optimizer = $ldaOptimizerOption")
 
+    // Set LDA params from input args
     val lda =
       new LDA()
         .setK(topicCount)
         .setMaxIterations(maxIterations)
         .setAlpha(ldaAlpha)
         .setBeta(ldaBeta)
-        .setOptimizer(optimizer)
+        .setOptimizer(ldaOptimizer)
 
     // If caller does not provide seed to lda, ie. ldaSeed is empty, lda is seeded automatically set to hash value of class name
 
     if (ldaSeed.nonEmpty) {
       lda.setSeed(ldaSeed.get)
     }
 
+    val (wordTopicMat, docTopicDist) = ldaOptimizer match {
+      case _: EMLDAOptimizer => {
+        val ldaModel = lda.run(ldaCorpus).asInstanceOf[DistributedLDAModel]
```

Comment:
I think this is already solved in the spark.ml library, for what I remember both implementations return the same model called just: LDAModel from the org.apache.spark.ml.clustering.